ngraph/src/ngraph/coordinate_transform.hpp

```diff
@@ -24,9 +24,48 @@
 
 namespace ngraph
 {
-    class NGRAPH_API CoordinateTransform
+    class NGRAPH_API CoordinateIterator
     {
     public:
+        CoordinateIterator(const Shape& target_shape, bool is_end = false);
+
+        void operator++();
+        CoordinateIterator operator++(int);
+        void operator+=(size_t n);
+        const Coordinate& operator*() const noexcept;
+        bool operator!=(const CoordinateIterator& it) const noexcept;
+        bool operator==(const CoordinateIterator& it) const noexcept;
+        void advance(size_t axis) noexcept;
+        static const CoordinateIterator& end();
+
+    private:
+        Shape m_target_shape;
+        Coordinate m_coordinate;
+        bool m_oob;
+        bool m_empty;
+    };
+
+    class NGRAPH_API CoordinateTransformBasic
```

Comment:
Please add documentation which explains when to use this class.

Reply:
fixed